scripts/generate_sitemap.py

```diff
@@ -0,0 +1,203 @@
+        tag_text = self.doc.createTextNode(text)
+        tag.appendChild(tag_text)
+
+    def add_url(self, config):
+        """Adds a url to the current urlset"""
+        if self.url_count >= settings.SITEMAP_URL_MAX:
+            self.write_doc()
+            self.new_doc()
+        self.url = self.doc.createElement('url')
+        self.urlset.appendChild(self.url)
+
+        for k, v in config.iteritems():
+            self.add_tag(k, v)
+        self.url_count += 1
+
+    def write_doc(self):
+        """Writes and gzips each sitemap xml file"""
+        path = os.path.join(self.sitemap_dir, 'sitemap_{}.xml'.format(str(self.sitemap_count)))
+        print('Writing and gzipping `{}`: url_count = {}'.format(path, str(self.url_count)))
+
+        xml_str = self.doc.toprettyxml(indent="  ", encoding='utf-8')
+        with open(path, 'w') as f:
+            f.write(xml_str)
+        # Write zipped file
+        with open(path, 'rb') as f_in, gzip.open(path + '.gz', 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out)
+        self.sitemap_count += 1
+
+    def write_sitemap_index(self):
+        """Writes the index file for all of the sitemap files"""
+        doc = xml.dom.minidom.Document()
+        sitemap_index = self.doc.createElement('sitemapindex')
+        sitemap_index.setAttribute('xmlns', 'http://www.sitemaps.org/schemas/sitemap/0.9')
+        doc.appendChild(sitemap_index)
+
+        for f in range(self.sitemap_count):
+            sitemap = doc.createElement('sitemap')
+            sitemap_index.appendChild(sitemap)
+
+            loc = doc.createElement('loc')
+            sitemap.appendChild(loc)
+            loc_text = self.doc.createTextNode(urlparse.urljoin(settings.DOMAIN, 'sitemaps/sitemap_{}.xml.gz'.format(str(f))))
+            loc.appendChild(loc_text)
+
+            datemod = doc.createElement('datemod')
+            sitemap.appendChild(datemod)
+            datemod_text = self.doc.createTextNode(datetime.datetime.now().isoformat())
+            datemod.appendChild(datemod_text)
+
+        print('Writing `sitemap_index.xml`')
+        xml_str = doc.toprettyxml(indent="  ", encoding='utf-8')
+        with open(os.path.join(self.sitemap_dir, 'sitemap_index.xml'), 'w') as f:
+            f.write(xml_str)
+
+    def log_errors(self, obj, obj_id, error):
+        if not self.errors:
+            script_utils.add_file_logger(logger, __file__)
+        self.errors += 1
+        logger.info('Error on {}, {}:'.format(obj, obj_id))
+        logger.exception(error)
+        if self.errors == 1000:
+            sentry.log_message('ERROR: generate_sitemap stopped execution after reaching 1000 errors. See logs for details.')
+            raise Exception('Too many errors generating sitemap.')
+
+    def generate(self):
+        print('Generating Sitemap')
+
+        # Progress bar
+        progress = Progress()
+
+        # Preprint urls
+        objs = PreprintService.find(Q('is_published', 'eq', True))
```

Comment:
This does not look like it's using the MODM-iterator to handle large data sets. Can you check in migrate_search for the example and make sure you're doing that? @erinspace has also done that recently with analytics.